client/web/src/components/popovers/ColorPicker.vue

```diff
@@ -115,13 +117,197 @@
 
 <script lang="ts">
 import { defineComponent } from "vue";
+import { clamp, HSV2RGB, RGB2HSV, RGB, HSV } from "../../lib/utils";
+
+interface Rect {
+	width: number;
+	height: number;
+	left: number;
+	top: number;
+}
+
+interface ColorPickerData {
+	color: HSV;
+	hue: {
+		rect: Rect;
+	};
+	saturation: {
+		rect: Rect;
+	};
+	opacity: {
+		rect: Rect;
+	};
+}
```

Comment:
I think it's probably sufficient to define `color`, `hue`, saturation`, and `opacity` on the Vue component's `data` object without having to add the complexity of wrapping it in an interface. If there was a reason to use a custom property instead of the `data` object, disregard this.

Reply:
Data is for reactive values, this is the way in Vue to add non-reactive data to a component

Reply:
https://vuejs.org/v2/api/#vm-options

Reply:
Alternatively, we can put the object in data() and prefix with an _, VueJS does not proxify object with underscore.
https://v3.vuejs.org/api/options-data.html#data-2